File: src/com/gaolei/java/MyStack.java
package com.gaolei.java;

import java.util.LinkedList;
import java.util.Queue;

class MyStack {
    private Queue<Integer> queue1 = new LinkedList<>();
    private Queue<Integer> queue2 = new LinkedList<>();

    /**
     * Initialize your data structure here.
     */
    public MyStack() {

    }

    /**
     * Push element x onto stack.
     */
    public void push(int x) {
        queue1.offer(x);
        while (!queue2.isEmpty()) {
            queue1.offer(queue2.poll());
        }
        Queue temp = queue1;
        queue1 = queue2;
        queue2 = temp;
    }

    /**
     * Removes the element on top of the stack and returns that element.
     */
    public int pop() {
        return queue2.poll();
    }

    /**
     * Get the top element.
     */
    public int top() {
        return queue2.peek();
    }

    /**
     * Returns whether the stack is empty.
     */
    public boolean empty() {
        return queue2.isEmpty();
    }
}

File: src/com/gaolei/kotlin/MyStack.kt
package com.gaolei.kotlin

import java.util.*

class MyStack() {

    /** Initialize your data structure here. */
    var queue1: Queue<Int> = LinkedList<Int>()
    var queue2: Queue<Int> = LinkedList<Int>()

    /** Push element x onto stack. */
    fun push(x: Int) {
        queue1.offer(x)
        while (queue2.isNotEmpty()) {
            queue1.offer(queue2.poll())
        }
        var temp = queue1
        queue1 = queue2
        queue2 = temp
    }

    /** Removes the element on top of the stack and returns that element. */
    fun pop(): Int {
        return queue2.poll()
    }

    /** Get the top element. */
    fun top(): Int {
        return queue2.peek()
    }

    /** Returns whether the stack is empty. */
    fun empty(): Boolean {
        return queue2.isEmpty()
    }

}
